dapr_agents/agents/components.py

```diff
@@ -67,38 +98,80 @@ def __init__(
         """
         self.name = name
         self._workflow_grpc_options = workflow_grpc_options
+        self._runtime_secrets: Dict[str, str] = {}
+        self._runtime_conf: Dict[str, str] = {}
 
+        # if pubsub is None or state is None or registry is None:
         with DaprClient() as _client:
             resp: GetMetadataResponse = _client.get_metadata()
             self.appid = resp.application_id
-            if pubsub is None or state is None or registry is None:
-                components: Sequence[RegisteredComponents] = resp.registered_components
-                for component in components:
-                    if (
-                        "state" in component.type
-                        and component.name == "agent-wfstatestore"
-                        and state is None
-                    ):
-                        state = AgentStateConfig(
-                            store=StateStoreService(store_name=component.name),
-                            state_key=f"{name.replace(' ', '-').lower() if name else 'default'}:workflow_state",
+            components: Sequence[RegisteredComponents] = resp.registered_components
+            for component in components:
+                if (
+                    "state" in component.type
+                    and component.name == "agent-wfstatestore"
+                    and state is None
+                ):
+                    state = AgentStateConfig(
+                        store=StateStoreService(store_name=component.name),
+                        state_key=f"{name.replace(' ', '-').lower() if name else 'default'}:workflow_state",
+                    )
+                if (
+                    "state" in component.type
+                    and component.name == "agent-registry"
+                    and registry is None
+                ):
+                    registry = AgentRegistryConfig(
+                        store=StateStoreService(store_name="agent-registry"),
+                        team_name="default",
+                    )
+                if "state" in component.type and component.name == "agent-runtime":
+                    raw_runtime_conf: StateResponse = _client.get_state(
+                        store_name=component.name,
+                        key="agent_runtime_configuration",
```

Comment:
if above we have agent-wfstatestore with key agent_workflow, then should we do agent-runtimestatestore with key agent_runtime?

Reply:
Fixed!